docs/advanced_topics.rst

```diff
@@ -1,114 +1,253 @@
+The ``max_real_memory_size`` value represents the biggest tensor in the batch for the outputs that
+allocate memory per sample and not for the entire batch at the time or the average tensor size when
+the allocation is contiguous. This value should be provided to ``bytes_per_sample_hint``.
 
-Prefetching queue depth
+Prefetching Queue Depth
 -----------------------
 
-The purpose of this functionality is to average the processing time between batches when the variation from batch to batch is high.
-DALI pipeline allows buffering one or more batches of data ahead. This becomes important when the data processing time between batches could vary a lot. Default prefetch depth is 2. The user can change this value using the ``prefetch_queue_depth`` pipeline argument. For example, if the variation is bigger then it is recommended to prefetch more ahead.
+When the variation between batches is high, this functionality averages the processing time between
+batches when the variation between batches is high.
+The DALI pipeline allows the buffering of one or more batches of data, which is important when
+the data processing time between batches can vary.
+The default prefetch depth value is 2. You can change this value by using the ``prefetch_queue_depth``
+pipeline argument. For example, if the variation is not hidden by the default prefetch depth value,
+we recommend that you prefetch more data ahead of time.
 
 Running DALI pipeline
 ---------------------
 
-DALI provides a couple of ways to run a pipeline:
+You can run a pipeline in DALI in one of the following ways:
+
+- | Simple run method, which runs the computations and returns the results.
+  | This option corresponds to the :meth:`nvidia.dali.types.PipelineAPIType.BASIC` API type.
+- | `Schedule_run`, `share_outputs`, and `release_outputs` that allows a fine-grain control for
+    the duration of the output buffers’ lifetime.
+  | This option corresponds to the :meth:`nvidia.dali.types.PipelineAPIType.SCHEDULED` API type.
+- | Built-in iterators for MXNet, PyTorch, and TensorFlow.
+  | This option corresponds to the :meth:`nvidia.dali.types.PipelineAPIType.ITERATOR` API type.
+
+The first API, :meth:`nvidia.dali.pipeline.Pipeline.run()` method completes the following tasks:
+
+#. Launches the DALI pipeline.
+#. Executes the prefetch iterations if necessary.
+#. Waits until the first batch is ready.
+#. Returns the resulting buffers.
+
+Buffers are marked as in-use until the next call to
+:meth:`nvidia.dali.pipeline.Pipeline.run`. This process can be wasteful because data is usually
+copied to the native framework tensors and then returned to DALI to be reused
```

Comment:
```suggestion
:meth:`nvidia.dali.pipeline.Pipeline.run`. This process can be wasteful because the data is usually
copied to the DL framework's native storage objects and DALI pipeline outputs could be returned to
DALI for reuse.
```

Reply:
Done